src/units/attack_type.hpp

```diff
@@ -87,12 +87,12 @@ class attack_type : public std::enable_shared_from_this<attack_type>
 	std::string weapon_specials() const;
 	std::string weapon_specials_value(const std::set<std::string> checking_tags) const;
 
-	/** Returns alignment specified by alignment_str_ variable If empty or not valid returns the unit's alignment or neutral if self_ variable empty.
+	/** Returns alignment specified by alignment_ variable.
 	 */
-	utils::optional<unit_alignments::type> alignment() const {return unit_alignments::get_enum(alignment_str_);}
+	utils::optional<unit_alignments::type> alignment() const { return alignment_; }
 	/** Returns alignment specified by alignment() for filtering when exist.
 	 */
-	std::string alignment_str() const;
+	std::string alignment_str() const { return alignment_ ? unit_alignments::get_string(*alignment()) : ""; }
```

Comment:
Is there any reason to just just directly use `alignment_`?

Reply:
the problem is that I use alignment_ and alignment() in the same line when I should use either one or the other, as we stay in the attack_type class I should use alignment_ directly without calling the intermediate function.

Reply:
@Pentarctagon what do you prefer, that I use alignment_ directly or that I go through alignment()?

Reply:
@Pentarctagon replaced alignment_ by alignement()

Reply:
I don't think there's any reason to use `alignment()` within the class itself when it can already use `alignment_` directly, unless I'm missing something. `alignment()` doesn't do anything else besides return `alignment_`.

Reply:
@Pentarctagon rechanged to alignment_ for all line